siriuspy/siriuspy/idff/main.py

```diff
@@ -0,0 +1,226 @@
+"""ID feedforward App."""
+
+import os as _os
+import logging as _log
+import time as _time
+import epics as _epics
+
+from ..callbacks import Callback as _Callback
+from ..devices import IDFF as _IDFF
+
+from .csdev import IDFFConst as _Const, ETypes as _ETypes
+
+
+class App(_Callback):
+    """Main application for handling machine shift."""
+
+    def __init__(self, idname):
+        """Class constructor."""
+        super().__init__()
+        self._const = _Const(idname)
+        self._pvs_prefix = self._const.idffname
+        self._pvs_database = self._const.get_propty_database()
+
+        self._loop_state = _Const.DsblEnbl.Dsbl
+        self._loop_freq = _Const.DEFAULT_LOOP_FREQ
+        self._control_qs = _Const.DsblEnbl.Dsbl
+        self._polarization = 'none'
+        self._config_name = ''
+        self.read_autosave_file()
+
+        # connections
+        # use pyepics recommendations for threading
+        _epics.ca.use_initial_context()
```

Comment:
do we need to invoke `use_initial_context` here? it is already being invoked within `CAThread.run`that is called in `CAThread.start`:

```Python3

class CAThread(threading.Thread):
    """
    Sub-class of threading.Thread to ensure that the
    initial CA context is used.
    """
    def run(self):
        use_initial_context()
        threading.Thread.run(self)

```

Reply:
I usually call use_initial_context before creating PVs in the App class, but maybe we don't need it

Reply:
done!